python/tests/test_highlevel.py

```diff
@@ -225,20 +228,23 @@ def get_gap_examples():
                 assert len(t.parent_dict) == 0
                 found = True
         assert found
-        yield ts
+        ret.append((f"gap {x}", ts))
     # Give an example with a gap at the end.
     ts = msprime.simulate(10, random_seed=5, recombination_rate=1)
     tables = get_table_collection_copy(ts.dump_tables(), 2)
     tables.sites.clear()
     tables.mutations.clear()
     insert_uniform_mutations(tables, 100, list(ts.samples()))
-    yield tables.tree_sequence()
+    ret.append(("gap at end", tables.tree_sequence()))
+    return ret
 
 
+@functools.lru_cache()
```

Comment:
Did you mean to cache this one and have it return a list, while `get_example_tree_sequences` doesn't have a cache decorator and most of the other build-ts-functions are still iterators?

Reply:
I converted the ones called from other places other than `get_example_tree_sequences`. `get_example_tree_sequences` is cached, but manually.